src/embedding_scvi/_module.py

```diff
@@ -216,10 +257,18 @@ def loss(
         X = tensors[REGISTRY_KEYS.X_KEY]
         posterior = inference_outputs[TENSORS_KEYS.QZ_KEY]
         prior = inference_outputs[TENSORS_KEYS.PZ_KEY]
-        likelihood = generative_outputs[TENSORS_KEYS.PX_KEY]
 
-        # (n_obs, n_latent) -> (n_obs,)
-        kl_div = dist.kl_divergence(posterior, prior).sum(dim=-1)
+        if self.prior=='mog' or self.prior=='mog_celltype':
+            u = posterior.rsample(sample_shape=(10,))
```

Comment:
Should we add the sample shape as a parameter?

Reply:
Potentially, in mrVI we fixed it. We should then expose it through model.train.